osu.Game/Database/RealmArchiveModelImporter.cs

```diff
@@ -105,14 +105,25 @@ public async Task<IEnumerable<Live<TModel>>> Import(ProgressNotification notific
             }
 
             notification.Progress = 0;
-            notification.Text = $"{HumanisedModelName.Humanize(LetterCasing.Title)} import is initialising...";
 
             int current = 0;
 
             var imported = new List<Live<TModel>>();
 
             parameters.Batch |= tasks.Length >= minimum_items_considered_batch_import;
 
+            // A paused state could obviously be entered mid-import (during the `Task.WhenAll` below),
+            // but in order to keep things simple let's focus on the most common scenario.
+            notification.Text = $"{HumanisedModelName.Humanize(LetterCasing.Title)} import is paused due to gameplay...";
+
+            try
+            {
+                pauseIfNecessary(parameters, notification.CancellationToken);
```

Comment:
It's a little bit scary that this is the first entrypoint to the async context (there's no `await` prior to this line). For example if `Import` is ever called from the update/bdl thread, then this is going to potentially deadlock the game.

I suggest either `await Task.Yield()` before the pause, or wrapping the entire thing in a `Task.Run()`.

Reply:
It's a fair point. I started by duplicating the method and making an async version, but then ended up refactoring things a bit to not require this.

Of note, the `Parallel.ForEachAsync` change was required because without it, the previous `Select` loop would seem to cause all imports to get past the pause check and run up to the `Import` part which would mean they never actually pause.

The whole import async logic is a bit of a mess but as far as I can tell it's working as expected.